source/Preprocessor.cpp

```diff
@@ -472,6 +474,68 @@ Trivia Preprocessor::handleDefaultNetTypeDirective(Token directive) {
     return Trivia(TriviaKind::Directive, result);
 }
 
+Trivia Preprocessor::handleUndefDirective(Token directive) {
+    Token nameToken = expect(TokenKind::Identifier);
+    StringRef name = nameToken.valueText();
```

Comment:
If name is empty (because nameToken was missing) we shouldn't also error about not finding the directive.